stat_test.go

```diff
@@ -836,33 +837,89 @@ var linearRegressionTests = []struct {
 
 		tol: 1e-5,
 	},
+	{
+		name: "x, y length mismatch",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: make([]float64, 3),
+	},
+	{
+		name: "weights mismatch with x",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 3),
+		weights: make([]float64, 2),
+	},
+	{
+		name: "x, y length mismatch and weights nil",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: nil,
+	},
+	{
+		name:    "weights are nil",
+		x:       []float64{12.0, 24.0, 21.0},
+		y:       []float64{1.0, 2.0, 3.0},
+		weights: nil,
+	},
 }
 
 func TestLinearRegression(t *testing.T) {
 	for _, test := range linearRegressionTests {
-		alpha, beta := LinearRegression(test.x, test.y, test.weights, test.origin)
-		var r float64
-		if test.origin {
-			r = RNoughtSquared(test.x, test.y, test.weights, beta)
+		if strings.Contains(test.name, "mismatch") {
```

Comment:
Rather than do this strings.Contains, just add a `panics bool` field to linearRegressionTest.

Then you can do
var alpha, beta float64
if !Panics(stuff) && test.panics {
     t.Errorf("LinearRegression unexpected success")
    continue
}
````
And then the tests that are there now.